### PR TITLE
ICAL.Event now has setters, new component updatePropertyWithValue

### PR DESCRIPTION
**Pull request by lightsofapollo**
I find that I like this layered approach to creating events. There are a lot of details (more to cover in here) when creating or editing a resource that we can't (or should not) cover in the components that is easily done in a wrapper layer.

Let me know what you think

### CODE REVIEW

**Comment by lightsofapollo** [comment]
@kewisch R?

File: lib/ical/event.js [hunk 1]
```diff
@@ -1,7 +1,19 @@
 ICAL.Event = (function() {
 
   function Event(component, options) {
-    this.component = component;
+    if (!(component instanceof ICAL.icalcomponent)) {
+      options = component;
+      component = null;
+    }
+
+    if (!component) {
```
Comment (kewisch) [inline lib/ical/event.js]: Just a suggestion, but I tend to turn around the if blocks here, i.e

```
if (component) {

} else {

}
```

This way JS avoids the ! operation. I don't know if its optimized out anyway, but doesn't hurt to try :)

**Comment by kewisch** [comment]
r=philipp on this. The comment on if block order is only minor, so if you like feel free to merge as it is.

**Comment by lightsofapollo** [comment]
@kewisch Good suggestion review comment addressed & landing.

**Pull request merged**